### PR TITLE
native token cross chain

### PR DESCRIPTION
**Pull request by dylenfu**

### CODE REVIEW

File: contracts/native/cross_chain_manager/common/param.go [hunk 1]
```diff
@@ -273,3 +294,35 @@ func (this *MultiSignParam) Deserialization(source *polycomm.ZeroCopySource) err
...
 	return nil
 }
+
+type TxArgs struct {
+	ToAssetHash []byte
+	ToAddress   []byte
+	Amount      *big.Int
+}
+
+func (this *TxArgs) Serialization(sink *polycomm.ZeroCopySink) {
```
Comment (siovanus) [inline contracts/native/cross_chain_manager/common/param.go:304]: we use zero copy or rlp, or same with evm?

File: contracts/native/cross_chain_manager/entrance.go [hunk 2]
```diff
@@ -98,142 +99,127 @@ func Name(s *native.NativeContract) ([]byte, error) {
...
 	params := &scom.EntranceParam{}
 	if err := utils.UnpackMethod(scom.ABI, scom.MethodImportOuterTransfer, params, ctx.Payload); err != nil {
 		return nil, err
 	}
 
-	chainID := params.SourceChainID
-	blacked, err := CheckIfChainBlacked(native, chainID)
+	srcChainID := params.SourceChainID
+	if native.IsMainChain(srcChainID) {
+		return nil, nil
```
[the remark this answers is not in the document]
Comment (dylenfu) [reply contracts/native/cross_chain_manager/entrance.go]: fixed

File: contracts/native/cross_chain_manager/common/param.go (hunk 1 above)
Comment (dylenfu) [reply contracts/native/cross_chain_manager/common/param.go]: replaced with rlp serialization now

**Review by siovanus — APPROVED** [review]

File: contracts/native/cross_chain_manager/eth/internal.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+		return err
+	}
+	var s []byte
+	for i := len(s_temp); i < 32; i++ {
+		s = append(s, 0)
+	}
+	s = append(s, s_temp...)
+
+	value = cacheDBRecover(value)
+	valueHash := common.BytesToHash(value)
```
Comment (siovanus) [inline contracts/native/cross_chain_manager/eth/internal.go:125]: value is raw byte not hash

**Pull request merged**